sdk/src/trace/tracer.cc

```diff
@@ -38,14 +40,20 @@ nostd::unique_ptr<trace_api::Span> Tracer::StartSpan(
       sampler_->ShouldSample(nullptr, trace_api::TraceId(), name, options.kind, attributes);
   if (sampling_result.decision == Decision::NOT_RECORD)
   {
-    return nostd::unique_ptr<trace_api::Span>{new (std::nothrow)
-                                                  trace_api::NoopSpan{this->shared_from_this()}};
+    auto span = nostd::shared_ptr<trace_api::Span>{
+        new (std::nothrow) trace_api::NoopSpan{this->shared_from_this()}};
+
+    return span;
   }
   else
   {
-    auto span = nostd::unique_ptr<trace_api::Span>{new (std::nothrow) Span{
+    auto span = nostd::shared_ptr<trace_api::Span>{new (std::nothrow) Span{
         this->shared_from_this(), processor_.load(), name, attributes, options}};
 
+    span->SetToken(
+        nostd::unique_ptr<context::Token>(new context::Token(context::RuntimeContext::Attach(
+            context::RuntimeContext::GetCurrent().SetValue(SpanKey, span)))));
```

Comment:
According to my understanding, we mustn't do this here. The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span-creation) says:
> Span creation MUST NOT set the newly created Span as the currently active Span by default, but this functionality MAY be offered additionally as a separate operation.

I think with doing this here, we cannot offer the functionality to create a span that is _not_ the currently active span. Should this rather be a separate operation on the span (`MarkActive`, or `SetActive`)?

Reply:
@pyohannes that's correct, in Python there are separate APIs which allows one to start span, with or without making it "current" to the tracer.
https://github.com/open-telemetry/opentelemetry-python/blob/76b1ed0d6f4d41df0075953dcfbece72dd4bb344/opentelemetry-api/src/opentelemetry/trace/__init__.py#L301

Reply:
I will put up a PR for this.